### PR TITLE
 - Issue #6

### PR DESCRIPTION
**Pull request by afabiani**
Added the possibility of specifying a target store and async mode on Importer sessions.

This feature is necessary if you need, as an instance, to add new granules to an existing ImageMosaic in GeoServer.

The GeoServer Importer supports the update of already existent stores since version 2.7.x.

### CODE REVIEW

File: src/gsimporter/api.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def parse_response(args, parent=None):
     '''
     headers, response = args
     try:
+        print (str(response))
```
Comment (simboss) [inline src/gsimporter/api.py]: Use logging properly.

**Comment by jj0hns0n** [comment]
I haven't run these tests in a very long time, and I dont think they are in a CI anywhere, but I get 3 failures  running against the GeoServer 2.7-SNAPSHOT from build.geonode.org

```
{"import":{"id":120,"href":"http://localhost:8080/geoserver/rest/imports/120","state":"PENDING","archive":false,"tasks":[]}}
ERROR

======================================================================
ERROR: test_mosaic (test.uploadtests.SingleImportTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 345, in test_mosaic
    layer_name = self.run_single_upload(raster=zip_file, mosaic=True)
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 221, in run_single_upload
    session = client.upload(file_name, mosaic=mosaic)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 77, in upload
    return self.upload_files(files, use_url, import_id, mosaic, initial_opts)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 99, in upload_files
    session.upload_task(files, use_url, intial_opts)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/api.py", line 402, in upload_task
    resp = self._client().put_zip(addopts(url), files[0])
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 187, in put_zip
    "Content-type": "application/zip",
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 162, in _request
    log_data = data if len(data) < 1024 else "[Data to long...]"
TypeError: object of type 'file' has no len()

======================================================================
ERROR: test_single_raster_upload_change_workspace (test.uploadtests.SingleImportTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 194, in tearDown
    gscat.delete(lyr.resource)
  File "build/bdist.macosx-10.9-intel/egg/geoserver/catalog.py", line 88, in delete
    rest_url = config_object.href
AttributeError: 'NoneType' object has no attribute 'href'

======================================================================
ERROR: test_upload_with_bad_files (test.uploadtests.SingleImportTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 329, in test_upload_with_bad_files
    expected_layer='importer:san_andres_y_providencia_poi')
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 221, in run_single_upload
    session = client.upload(file_name, mosaic=mosaic)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 77, in upload
    return self.upload_files(files, use_url, import_id, mosaic, initial_opts)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 99, in upload_files
    session.upload_task(files, use_url, intial_opts)
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/api.py", line 402, in upload_task
    resp = self._client().put_zip(addopts(url), files[0])
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 187, in put_zip
    "Content-type": "application/zip",
  File "/Users/ortelius/projects/gsimporter/src/gsimporter/client.py", line 162, in _request
    log_data = data if len(data) < 1024 else "[Data to long...]"
TypeError: object of type 'file' has no len()

----------------------------------------------------------------------
Ran 18 tests in 5.179s

FAILED (errors=3, skipped=1)
```

Also, note I had to create the importer DB as the error message mentioned. We should probably add this to the README.

```
Traceback (most recent call last):
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 63, in <module>
    conn = open_db_datastore_connection()
  File "/Users/ortelius/projects/gsimporter/test/uploadtests.py", line 61, in open_db_datastore_connection
    return psycopg2.connect(' '.join([ "%s='%s'" % (k, DB_CONFIG[v]) for k,v in params ]))
  File "/Users/ortelius/projects/gsimporter/psycopg2-2.6.1-py2.7-macosx-10.9-intel.egg/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
OperationalError: FATAL:  role "importer" does not exist

Error connecting to the database, check your settings
{'DB_DATASTORE_DATABASE': 'importer_test',
 'DB_DATASTORE_HOST': 'localhost',
 'DB_DATASTORE_NAME': 'importer_test',
 'DB_DATASTORE_PASSWORD': 'importer',
 'DB_DATASTORE_PORT': '5432',
 'DB_DATASTORE_TYPE': 'postgis',
 'DB_DATASTORE_USER': 'importer'}
```

**Comment by simboss** [comment]
I would require tests to be added  before merging.

**Comment by afabiani** [comment]
Dear all,
I created a new PR which:

1. fixes all the uploadtests
2. has a test which demonstrates the Importer capability of adding a new
granule to an existing mosaic
3. fixes alla the pep8 violations of uploadtests.py

https://github.com/boundlessgeo/gsimporter/pull/7

kind regards,
Alessio


==
GeoServer Professional Services from the experts! Visit
http://goo.gl/NWWaa2 for more information.
==

Ing. Alessio Fabiani
@alfa7691
Founder/Technical Lead

GeoSolutions S.A.S.
Via Poggio alle Viti 1187
55054  Massarosa (LU)
Italy
phone: +39 0584 962313
fax:     +39 0584 1660272
mob:   +39  331 6233686

http://www.geo-solutions.it
http://twitter.com/geosolutions_it

-------------------------------------------------------

*AVVERTENZE AI SENSI DEL D.Lgs. 196/2003*

Le informazioni contenute in questo messaggio di posta elettronica e/o
nel/i file/s allegato/i sono da considerarsi strettamente riservate. Il
loro utilizzo è consentito esclusivamente al destinatario del messaggio,
per le finalità indicate nel messaggio stesso. Qualora riceviate questo
messaggio senza esserne il destinatario, Vi preghiamo cortesemente di
darcene notizia via e-mail e di procedere alla distruzione del messaggio
stesso, cancellandolo dal Vostro sistema. Conservare il messaggio stesso,
divulgarlo anche in parte, distribuirlo ad altri soggetti, copiarlo, od
utilizzarlo per finalità diverse, costituisce comportamento contrario ai
principi dettati dal D.Lgs. 196/2003.



The information in this message and/or attachments, is intended solely for
the attention and use of the named addressee(s) and may be confidential or
proprietary in nature or covered by the provisions of privacy act
(Legislative Decree June, 30 2003, no.196 - Italy's New Data Protection
Code).Any use not in accord with its purpose, any disclosure, reproduction,
copying, distribution, or either dissemination, either whole or partial, is
strictly forbidden except previous formal approval of the named
addressee(s). If you are not the intended recipient, please contact
immediately the sender by telephone, fax or e-mail and delete the
information in this message that has been received in error. The sender
does not give any warranty or accept liability as the content, accuracy or
completeness of sent messages and accepts no responsibility  for changes
made after they were sent or for other risks which arise as a result of
e-mail transmission, viruses, etc.

**Comment by afabiani** [comment]
Edited the title of the PR (
https://github.com/boundlessgeo/gsimporter/pull/7)

==
GeoServer Professional Services from the experts! Visit
http://goo.gl/NWWaa2 for more information.
==

Ing. Alessio Fabiani
@alfa7691
Founder/Technical Lead

GeoSolutions S.A.S.
Via Poggio alle Viti 1187
55054  Massarosa (LU)
Italy
phone: +39 0584 962313
fax:     +39 0584 1660272
mob:   +39  331 6233686

http://www.geo-solutions.it
http://twitter.com/geosolutions_it

-------------------------------------------------------

*AVVERTENZE AI SENSI DEL D.Lgs. 196/2003*

Le informazioni contenute in questo messaggio di posta elettronica e/o
nel/i file/s allegato/i sono da considerarsi strettamente riservate. Il
loro utilizzo è consentito esclusivamente al destinatario del messaggio,
per le finalità indicate nel messaggio stesso. Qualora riceviate questo
messaggio senza esserne il destinatario, Vi preghiamo cortesemente di
darcene notizia via e-mail e di procedere alla distruzione del messaggio
stesso, cancellandolo dal Vostro sistema. Conservare il messaggio stesso,
divulgarlo anche in parte, distribuirlo ad altri soggetti, copiarlo, od
utilizzarlo per finalità diverse, costituisce comportamento contrario ai
principi dettati dal D.Lgs. 196/2003.



The information in this message and/or attachments, is intended solely for
the attention and use of the named addressee(s) and may be confidential or
proprietary in nature or covered by the provisions of privacy act
(Legislative Decree June, 30 2003, no.196 - Italy's New Data Protection
Code).Any use not in accord with its purpose, any disclosure, reproduction,
copying, distribution, or either dissemination, either whole or partial, is
strictly forbidden except previous formal approval of the named
addressee(s). If you are not the intended recipient, please contact
immediately the sender by telephone, fax or e-mail and delete the
information in this message that has been received in error. The sender
does not give any warranty or accept liability as the content, accuracy or
completeness of sent messages and accepts no responsibility  for changes
made after they were sent or for other risks which arise as a result of
e-mail transmission, viruses, etc.

**Pull request closed without merging**